### PR TITLE
Contributing Confluence search functionality

### PR DESCRIPTION
**Pull request by su-gupta**
Signed-off-by: su-gupta <sugupta@brex.com>

## Hey, I just made a Pull Request!

Adding the Confluence Search functionality we developed (collator + Item Listing) to contrib so anyone can use these as a starting point to add Confluence to their own search.

Caveat: this was created for a proof of concept, so it's definitely not optimized (i.e. API requests are serial) but it was able to load all our documents in a matter of minutes - please feel free to use this as a starting point to build upon

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: contrib/search/confluence/ConfluenceCollator [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { DocumentCollator } from '@backstage/search-common';
```
Comment (iamEAP) [inline contrib/search/confluence/ConfluenceCollator:1]: Perhaps worth adding a `.ts` to the filename here?  Could make it easier to ready (syntax highlighting) when browsing.

File: contrib/search/confluence/ConfluenceResultListItem [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
```
Comment (iamEAP) [inline contrib/search/confluence/ConfluenceResultListItem:1]: Same here (`.tsx`)

**Review by iamEAP — APPROVED** [review]
Minor comments from me on the envelope for these contributions -- I didn't review the code itself because I don't really have a way to test, but trust it works.

File: contrib/search/confluence/ConfluenceCollator (hunk 1 above)
Comment (su-gupta) [reply contrib/search/confluence/ConfluenceCollator:1]: Hey @iamEAP! Good call, I actually originally had the file types included but I repeatedly ran into an eslint error 
`  1:1  error  Parsing error: The keyword 'import' is reserved`
I tried doing a little debugging and updating my .eslintrc.js file but unfortunately couldn't get it work and don't have time to look into it further

Comment (su-gupta) [reply contrib/search/confluence/ConfluenceCollator:1]: Let me know if there is a known fix for this, if so I'd be happy to update

Comment (jrusso1020) [reply contrib/search/confluence/ConfluenceCollator:1]: From taking a quick look, it looks like the contrib folder is mostly markdown files with code snippets

So maybe another workaround is having all the code as code snippets in a markdown file? You can then add the name of the files as part of the markdown file in the example

File: contrib/search/confluence/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Confluence
+
+These files help you add Confluence as a source to the Backstage Search plugin.
+To do so, add both files in this directory under the packages/backend/src/plugins/search/ pathway in your Backstage app as TypeScript files.
+Then, update your packages/app/src/components/search/SearchPage.tsx and packages/backend/src/plugins/search.ts
+to include the new Search source.
```
Comment (emmaindal) [inline contrib/search/confluence/README.md:6]: Do you want to add two examples here of how to include it. 

For example in `packages/backend/src/plugins/search.ts` something like:

```ts
indexBuilder.addCollator({
  defaultRefreshIntervalSeconds: 600,
  collator: // instantiate ConfluenceCollator
});
```

and for `packages/app/src/components/search/SearchPage.tsx` something like:

```tsx
case 'confluence':
            return (
              < ConfluenceResultListItem
                key={result.document.location}
                result={result.document}
              />
            );
          // ...
```

File: contrib/search/confluence/ConfluenceCollator (hunk 1 above)
Comment (su-gupta) [reply contrib/search/confluence/ConfluenceCollator]: Thanks for the callout - updated to markdown files!

File: contrib/search/confluence/README.md (hunk 3 above)
Comment (su-gupta) [reply contrib/search/confluence/README.md]: Good call, added code snippets to the README!

**Review by emmaindal — APPROVED** [review]
Awesome!! Thank you so much for the contribution!

**Comment by emmaindal** [comment]
@backstage/maintainers need your review too!

**Pull request merged**